src/main/java/io/cdap/plugin/http/source/common/http/OAuthUtil.java

```diff
@@ -54,5 +67,52 @@ public static String getAccessTokenByRefreshToken(CloseableHttpClient httpclient
     JsonElement jsonElement = JSONUtil.toJsonObject(responseString).get("access_token");
     return jsonElement.getAsString();
   }
+
+  public static String getAccessTokenByServiceAccount(BaseHttpSourceConfig config) throws IOException {
+    GoogleCredentials credential;
+    String jwtToken = "";
+    if (config.isServiceAccountJson()) {
+      InputStream jsonInputStream = new ByteArrayInputStream(config.getServiceAccountJson().getBytes());
+      credential = GoogleCredentials.fromStream(jsonInputStream);
+      jwtToken = generateJwt(5000, credential);
```

Comment:
How did you decide on 5 seconds? Also it's better to have it at least as a constant as it makes code more readable

Reply:
This is 5000 seconds based on the documentation. I'm not sure what time range I should choose

Reply:
Oh, I see, you convert it to millis. 5000 seconds is probably fine, but please make a constant that clearly says it's seconds in it's name

Reply:
Also 5000 seconds is an odd time. I would expect it to be aligned to an hour (3600 seconds) or a day (86400 seconds).

I also think this value should be in a configuration option for the user to configure. Some users might not feel comfortable with long living JWT tokens.